### PR TITLE
emails: handle long personal emails more gracefully

### PR DESCRIPTION
**Pull request by sjbarag**
**Reviewer:** @shakyShane 
**Asana:** https://app.asana.com/0/0/1206883124575876/f

## Description
Ellipsis-ize long @duck.com emails when they overflow their containers.

## Steps to test
1. Sign up for an `@duck.com` email with a pathologically long address, or [apply this patch instead](https://gist.github.com/sjbarag/dfabe15dd24696a66821cba9db5750bf)
2. Load any signup page (I used https://signup.live.com/signup)
3. Resize the page, then dismiss and re-show the autofill popup
    * ⚠️ We seem to have a separate bug that causes an infinite loop of resize callbacks when the page width crosses below 300px. That repro's on `main` at the moment, so it's not introduced or made worse by this PR. But yeah, just be aware of that I guess.

### Extension
image

### macOS
image

### CODE REVIEW

File: src/UI/EmailHTMLTooltip.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import { formatDuckAddress, escapeXML } from '../autofill-utils.js'
 import HTMLTooltip from './HTMLTooltip.js'
 
+/** A zero-width space literal. Some editors render it as a regular space. */
+const ZWSP = '​';
+
```
Comment (shakyShane) [inline src/UI/EmailHTMLTooltip.js:6]: given how this looks in GH and certain editors, what about using a unicode escape sequence instead? I think it should work, but I havn't checked :)

**Review by shakyShane — COMMENTED** [review]
@sjbarag hey! thanks for this - it's a nice improvement whilst also being a useful way to onboard you to the project 💪🏻 

Your changes will only impact the extension though I believe? so, we could use this as an opportunity to help you get setup on macOS too (since any change like this will need testing on desktop).

^ if that's a blocker still (waiting for the apple developer account) - then another thing you can do in the mean time is take a look at adding a new integration test for this change. 

Let me know if I can help in any way :)

**Comment by sjbarag** [comment]
Good call! I'll get right on both of those 😀

File: src/UI/EmailHTMLTooltip.js (hunk 1 above)
Comment (sjbarag) [reply src/UI/EmailHTMLTooltip.js:6]: Can-do!

**Comment by sjbarag** [comment]
Integration tests added — going to try this on macOS now ⏳

**Comment by sjbarag** [comment]
Okiedokes, this is ready for another round. Per @GioSensation's recommendation in Asana, we're not just truncating email addresses with an ellipsis. Working in both the extension and the native app, but I think I'll need to write separate integration tests for the native app still.

**Pull request closed without merging**